docs/book/v3/reference/migration.md

```diff
@@ -18,6 +18,14 @@ previously might have worked will now raise a `TypeError`.
 The following signatures changed that could affect class extension and/or
 consumers.
 
+### Interface suffix
+
+As of
+[laminas-coding-standard](https://github.com/laminas/laminas-coding-standard)
+version `2.0`, interface names must have an `Interface` suffix. The `Server`
+interface has been renamed to `ServerInterface` and the `Client` interface to
+`ClientInterface`.
+
```

Comment:
We should probably introduce the `Interface` suffix variants in a new minor release on the 2.X series, and have the old, non-suffixed versions extend them. This will allow deprecation.

Ideally, we should also provide a laminas-cli command in this package to help users update any implementations they have to the new variants; that should be done in the 2.x series as well.

Reply:
The new interfaces will be introduced in 2.9.

For the command to work, one would execute the command, updating userland code to the usage of the new interfaces, right? That would mean, a minor update to make the command available and immediately upgrade to 3.x, because the new interfaces are not accepted in 2.x as arguments. To me it would make more sense to upgrade directly to 3.x and execute the command. Maybe I am missing the value of an intermediate minor update(?).

Reply:
I got it backwards: have the _new_ interfaces extend the _old_interfaces.

We did this in laminas-eventmanager and laminas-servicemanager going into v3. By introducing them in v2 releases, users were able to update their code to implement the new interfaces, while keeping backwards compatibility. Then, when they upgraded to version 3, they did not need to make any further changes.

Most of those consumers are Laminas packages. This essentially allows them to pin to **either** the 2.X series **or** the 3.X series, if they update their interfaces.

Now that we have a stable release of laminas-cli (stable in that there is a version released that you can pin to semantically in order to keep a stable API — we dropped version 0.1.0 this morning), you can even go one step further: provide a tool to help the user migrate their implementations to the new interfaces. Something like `laminas server:migrate:v3-interfaces .`, which would go through the current directory, find implementations, and update them.

Of course: all of this is moot if the v2 and v3 interfaces are not compatible. If that's the case, then scratch the whole thing, and instead just provide a command in v3 for helping migrate to the new interfaces (or have us supply [Rector configuration](https://getrector.org/) for doing that).

Reply:
> I got it backwards: have the new interfaces extend the _old_interfaces.

That would have been nice to know in #8.

Reply:
PR with changes for 2.9 is provided. I opened an issue for a possible migration command (see #11).